### PR TITLE
Features Enhancement as in Issue #14

### PR DESCRIPTION
**Pull request by imabp**
## Changes
- Created a Navbar ( Issue #14 )
- Added Blog Section ( Issue #14 )
- Added Sections for Experience, Projects, and  Education ( Issue #14 ) by @amansharma2910 
- Implemented a complete Blog section with both features, as a custom blog or custom link ( Issue #14 ) by @ksdkamesh99
- @ksdkamesh99 created about.MD file to update the info on the homepage.
- A Custom 404 page along with navigation ( Issue #49 )
- Fixed a bug, in the image display.
## How Has This Been Tested?

We have run it locally and viewed every routes and components. 
We are attaching few screenshots for your reference.

image

## Mentions
@ksdkamesh99 : For his efforts in making blogs implementation and about section. :)
@amansharma2910 : For his efforts in making templates for different sections like experience and feeding data. 

## Types of changes

- [x] New feature and improvements (non-breaking change which adds/improves functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] Coding style of this project is followed

### CODE REVIEW

**Comment by Korusuke** [comment]
@imabp @amansharma2910 @ksdkamesh99 Thanks for this amazing PR!

I just have a few concerns:
- Currently we have a one page portfolio where in it's easy to get an overview of everything quickly and also some features like PDF generation works only with this as of now.
- In the new design, on home page there seems to be two navbars (1 on top serving as a pseudo navbar with just resume and 1 being the actual one)

I myself don't mind multi page portfolio but would prefer one pager, is it possible to have config for this instead of having any of them compulsory?

@wrussell1999 any views on this?

**Comment by amansharma2910** [comment]
@Korusuke it's a valid concern. But I think the previous single page portfolio was in a way restricted. The owner couldn't make a lot of changes to the home page due to the restricted sections. Now, since the about me section on the front page is a .md file, the owner can make fun customizations to their home page while keeping the other sections organized and intact.

**Comment by imabp** [comment]
> @imabp @amansharma2910 @ksdkamesh99 Thanks for this amazing PR!
> 
> I just have a few concerns:
> 
> * Currently we have a one page portfolio where in it's easy to get an overview of everything quickly and also some features like PDF generation works only with this as of now.
> * In the new design, on home page there seems to be two navbars (1 on top serving as a pseudo navbar with just resume and 1 being the actual one)
> 
> I myself don't mind multi page portfolio but would prefer one pager, is it possible to have config for this instead of having any of them compulsory?
> 
> @wrussell1999 any views on this?
Glad to see you sir @Korusuke, 😁

As I saw resume logic [here](https://github.com/MLH-Fellowship/portfolio-template/blob/15912fe030eac079fca7a7f2b0cb0c44f5e24db3/_includes/navbar.html#L6), it was actually, a pdf being downloaded from the source file in `cv/resume.pdf`. Let me know if I need to correct myself here.

I thought that this won't be a good idea, instead, we have so many data files and templates, we can generate a resume, by putting up a custom page for the same, and use some tweaks via the existing data files and let user download that as resume.

I hope that makes sense. 🥺

**Comment by Korusuke** [comment]
> @Korusuke it's a valid concern. But I think the previous single page portfolio was in a way restricted. The owner couldn't make a lot of changes to the home page due to the restricted sections. Now, since the about me section on the front page is a .md file, the owner can make fun customizations to their home page while keeping the other sections organized and intact.

I agree on this part, so can we add something like a config variable to change b/w `single` and `multi` page portfolio?
and also add the about me section in single page view too?

> I thought that this won't be a good idea, instead, we have so many data files and templates, we can generate a resume, by putting up a custom page for the same, and use some tweaks via the existing data files and let user download that as resume.

I believe this is being worked on in #37

**Comment by imabp** [comment]
> I believe this is being worked on in #37

@Korusuke I got this thing as an approach to avoid resume conflicts.
And have informed @suhanichawla the same.
What are your views on the below approach?

For resume
- [ ] Prepare a different page for resume
- [ ] In this page, add the various sections we need as a single-page resume
- [ ] And there should be a download button at Top, which can help you to get the resume as a pdf.

**Comment by Korusuke** [comment]
> > I believe this is being worked on in #37
> 
> @Korusuke I got this thing as an approach to avoid resume conflicts.
> And have informed @suhanichawla the same.
> What are your views on the below approach?
> 
> For resume
> 
>     * [ ]  Prepare a different page for resume
> 
>     * [ ]  In this page, add the various sections we need as a single-page resume
> 
>     * [ ]  And there should be a download button at Top, which can help you to get the resume as a pdf.

Sounds good to me as long as the resume generation is done in compile time, I don't like the idea of it being in runtime as that opens door for editable resume.

File: _layouts/blog.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+layout: default
+---
+
+<div class="banner">
+    <div class="container">
+        <div class="logo">
+            <a href="{{ site.url }}">
```
Comment (jkmartindale) [inline _layouts/blog.html:8]:
```suggestion
            <a href="{{ '/' | absolute_url }}">
```

File: _layouts/blog.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                <img src="{{ site.url }}/assets/img/logo.svg"/>
```
Comment (jkmartindale) [inline _layouts/blog.html:9]:
```suggestion
                <img src="{{ '/assets/img/logo.svg' | absolute_url }}"/>
```

File: _layouts/section.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+layout: default
+---
+
+<div class="section-banner">
+    <div class="container">
+        <div class="logo">
+            <a href="{{ site.url }}">
```
Comment (jkmartindale) [inline _layouts/section.html:8]:
```suggestion
            <a href="{{ '/' | absolute_url }}">
```

File: _layouts/section.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                <img src="{{ site.url }}/assets/img/logo.svg"/>
```
Comment (jkmartindale) [inline _layouts/section.html:9]:
```suggestion
                <img src="{{ '/assets/img/logo.svg' | absolute_url }}"/>
```

**Review by jkmartindale — CHANGES_REQUESTED** [review]
Small fix I'd like to see:

`{{ site.url }}` isn't the right way to make absolute URLs in Jekyll. It works some of the time but can break. The proper way is to use `{{ '/my/relative/url' | absolute_url }}`, which will work in all cases.

**Comment by ksdkamesh99** [comment]
Thanks for suggesting the changes @jkmartindale. Will definetly change the suggested ones and commit it soon

**Pull request closed without merging**